starknet_py/utils/typed_data.py

```diff
@@ -40,6 +100,24 @@ def from_dict(data: TypedDataDict) -> "TypedData":
         """
         return cast(TypedData, TypedDataSchema().load(data))
 
+    def to_dict(self) -> dict:
```

Comment:
Yes, the reason it was done this way is because initially `TypedData`'s nested fields weren't dumped to dict correctly. However, the change from
```domain = fields.Dict(data_key="domain", required=True)```
to 
```domain = fields.Nested(DomainSchema, required=True)```
lets use rely on `dump` method 👍